server/sonar-server/src/main/java/org/sonar/server/computation/measure/Measure.java

```diff
@@ -171,4 +131,28 @@ public String toString() {
    */
   QualityGateStatus getQualityGateStatus();
 
+  /**
+   * Any Measure, which ever is its value type, can have a Variations.
+   */
+  boolean hasVariations();
+
+  /**
+   * The variations of this measure.
+   *
+   * @throws IllegalStateException if the measure has no MeasureVariations
+   */
+  MeasureVariations getVariations();
+
+  /**
+   * The optional description of the measure. Relevant for manual measures.
+   */
+  @CheckForNull
+  String getDescription();
+
+  /**
+   * a Metric is equal to another Metric if it has the same ruleId/characteristicId paar (both being potentially
+   * {@code null} but only one of them can be non {@code null}.
+   */
+  boolean equals(Object o);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)